modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/export/graphs/plugins/DOTContentGraphExport.java

```diff
@@ -0,0 +1,136 @@
+import org.nuxeo.apidoc.export.graphs.introspection.ContentGraphExport;
+
+/**
+ * Exporter for DOT graph format, relying on jgrapht library.
+ *
+ * @since 20.0.0
+ */
+public class DOTContentGraphExport extends ContentGraphExport {
+
+    public DOTContentGraphExport() {
+        super();
+    }
+
+    @Override
+    public void updateContent(GraphExport graph) {
+        init(graph);
+
+        SimpleDirectedGraph<IdNode, Edge> g = new SimpleDirectedGraph<>(Edge.class);
+
+        int itemIndex = 1;
+        Map<String, IdNode> idMap = new HashMap<>();
+        for (Node<?> node : graph.getNodes()) {
+            IdNode idNode = new IdNode(itemIndex, node);
+            g.addVertex(idNode);
+            idMap.put(node.getId(), idNode);
+            itemIndex++;
+        }
+
+        for (Edge edge : graph.getEdges()) {
+            Node<?> source = graph.getNode(edge.getSource());
+            Node<?> target = graph.getNode(edge.getTarget());
+            g.addEdge(idMap.get(source.getId()), idMap.get(target.getId()), edge);
+        }
+
+        try {
+            ComponentNameProvider<IdNode> vertexIDProvider = new ComponentNameProvider<>() {
+                @Override
+                public String getName(IdNode idNode) {
+                    return String.valueOf(idNode.getId());
+                }
+            };
+            ComponentNameProvider<IdNode> vertexLabelProvider = new ComponentNameProvider<>() {
+                @Override
+                public String getName(IdNode idNode) {
+                    return idNode.getNode().getLabel();
+                }
+            };
+            ComponentNameProvider<Edge> edgeLabelProvider = new ComponentNameProvider<>() {
+                @Override
+                public String getName(Edge edge) {
+                    return edge.getValue();
+                }
+            };
+            ComponentAttributeProvider<IdNode> vertexAttributeProvider = new ComponentAttributeProvider<>() {
+                public Map<String, String> getComponentAttributes(IdNode idNode) {
+                    var map = new LinkedHashMap<String, String>();
+                    Node<?> node = idNode.getNode();
+                    map.put("weight", String.valueOf(node.getWeight()));
+                    map.put("type", String.valueOf(node.getType()));
+                    map.putAll(node.getAttributes());
+                    return map;
+                }
+            };
+            DOTExporter<IdNode, Edge> exporter = new DOTExporter<>(vertexIDProvider, vertexLabelProvider,
+                    edgeLabelProvider, vertexAttributeProvider, null);
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            exporter.exportGraph(g, out);
+            setContent(out.toString());
+            setContentName("graph.dot");
+            setContentType("application/octet-stream");
+        } catch (ExportException e) {
+            throw new RuntimeException(e);
```

Comment:
Maybe throws a `NuxeoException`, `RuntimeServiceException`, or one of ours exception instead?